securetar/__init__.py

```diff
@@ -714,10 +725,14 @@ def __init__(
         self._mode = mode
         self._bufsize = bufsize
         self._fileobj = fileobj
-        self._password = password
         self._streaming = streaming
         self._tar: tarfile.TarFile | None = None
 
+        # Set up key context
+        self._root_key_context: SecureTarRootKeyContext = root_key_context
```

Comment:
`self._root_key_context` is optional unless we do some validation that either that or password is required.
```suggestion
        self._root_key_context = root_key_context
```